### ISSUE TITLE
App crashes on sync

### ISSUE THREAD

**Issue by tecufanujacu** [stefan-niedermann/nextcloud-notes#652]
Android 6.0.1
Nextcloud Notes 2.2.0
Nextcloud app 3.9.2

Nextcloud Notes continuously crashes, this has started happening with new Nextcloud app update.

This is the error:
```
[01-01 18:46:55.819 31167:31190 E/AndroidRuntime]
FATAL EXCEPTION: AsyncTask #2
Process: it.niedermann.owncloud.notes, PID: 31167
Theme: themes:{default=, fontPkg:org.mmk2410.cyngn.theme.fira}
java.lang.RuntimeException: An error occurred while executing doInBackground()
	at android.os.AsyncTask$3.done(AsyncTask.java:309)
	at java.util.concurrent.FutureTask.finishCompletion(FutureTask.java:354)
	at java.util.concurrent.FutureTask.setException(FutureTask.java:223)
	at java.util.concurrent.FutureTask.run(FutureTask.java:242)
	at android.os.AsyncTask$SerialExecutor$1.run(AsyncTask.java:234)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1113)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:588)
	at java.lang.Thread.run(Thread.java:818)
Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'int java.lang.String.length()' on a null object reference
	at org.json.JSONTokener.nextCleanInternal(JSONTokener.java:116)
	at org.json.JSONTokener.nextValue(JSONTokener.java:94)
	at org.json.JSONArray.<init>(JSONArray.java:92)
	at org.json.JSONArray.<init>(JSONArray.java:108)
	at it.niedermann.owncloud.notes.util.ServerResponse$NotesResponse.getNotes(ServerResponse.java:37)
	at it.niedermann.owncloud.notes.persistence.NoteServerSyncHelper$SyncTask.pullRemoteChanges(NoteServerSyncHelper.java:351)
	at it.niedermann.owncloud.notes.persistence.NoteServerSyncHelper$SyncTask.doInBackground(NoteServerSyncHelper.java:277)
	at it.niedermann.owncloud.notes.persistence.NoteServerSyncHelper$SyncTask.doInBackground(NoteServerSyncHelper.java:248)
	at android.os.AsyncTask$2.call(AsyncTask.java:295)
	at java.util.concurrent.FutureTask.run(FutureTask.java:237)
	... 4 more
```

**Comment by stefan-niedermann** [comment]
So you updated the Nextcloud app to version `3.9.2` and then the notes app stopped working?

Are you using any other app with SSO (Deck or News)?

cc @tobiasKaminsky

**Comment by tecufanujacu** [comment]
Yes, after updating Nextcloud app to 3.9.2 then notes app stopped working.
This has happened on three phones, two with Android 6.0.1 and one with Android 4.4.4.
No, I'm not using other apps with SSO but I'm using DAVx⁵ without problems.
I can tell you that at the moment I'm still using the old Nextcloud server 12.0.3 and by default SSO & SAML app is disabled.

**Comment by stefan-niedermann** [comment]
Okay. Have you been asked to import an account after upgrading Notes? There should have been a dialog where you can pick an account. Did this work in the first place?

**Comment by stefan-niedermann** [comment]
Can you check whether or not this is related to #653 ? (Open the files app and then check whether the notes app synchronizes well directly after that)

**Comment by tecufanujacu** [comment]
> Okay. Have you been asked to import an account after upgrading Notes? There should have been a dialog where you can pick an account. Did this work in the first place?

Yes, after installing a new version, at first start I have been asked to choice an account, after this the Notes app has also been able to login in Nexcloud and load all the notes but it hasn't been able to sync them and it crashed continuosly, also when left in background.



> Can you check whether or not this is related to #653 ? (Open the files app and then check whether the notes app synchronizes well directly after that)

I don't think they are related, the problem that i noticed happened to me on three phones and two of them doesn't have any authentication method for Nextcloud app, nor fingerprint nor pin/password.

**Comment by arcanemachine** [comment]
Also getting crashes... my Nextcloud is on my LAN only and I get crashes when I'm on another WiFi network (app is set to sync only on WiFi).

Xiaomi Redmi Note 5 running Android Pie, newest version of Notes (old one still works fine):

time: 1578012983723
msg: java.lang.NullPointerException: Attempt to invoke virtual method 'int java.lang.String.length()' on a null object reference
stacktrace: java.lang.RuntimeException: An error occurred while executing doInBackground()
	at android.os.AsyncTask$3.done(AsyncTask.java:354)
	at java.util.concurrent.FutureTask.finishCompletion(FutureTask.java:383)
	at java.util.concurrent.FutureTask.setException(FutureTask.java:252)
	at java.util.concurrent.FutureTask.run(FutureTask.java:271)
	at android.os.AsyncTask$SerialExecutor$1.run(AsyncTask.java:245)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
	at java.lang.Thread.run(Thread.java:764)
Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'int java.lang.String.length()' on a null object reference
	at org.json.JSONTokener.nextCleanInternal(JSONTokener.java:116)
	at org.json.JSONTokener.nextValue(JSONTokener.java:94)
	at org.json.JSONObject.<init>(JSONObject.java:159)
	at org.json.JSONObject.<init>(JSONObject.java:176)
	at it.niedermann.owncloud.notes.util.ServerResponse$NoteResponse.getNote(ServerResponse.java:26)
	at it.niedermann.owncloud.notes.persistence.NoteServerSyncHelper$SyncTask.pushLocalChanges(NoteServerSyncHelper.java:303)
	at it.niedermann.owncloud.notes.persistence.NoteServerSyncHelper$SyncTask.doInBackground(NoteServerSyncHelper.java:275)
	at it.niedermann.owncloud.notes.persistence.NoteServerSyncHelper$SyncTask.doInBackground(NoteServerSyncHelper.java:248)
	at android.os.AsyncTask$2.call(AsyncTask.java:333)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	... 4 more


Running older versions of Nextcloud and Notes (haven't updated since install about 9 months ago. Trying to find the currently installed versions (seems to be a PITA on mobile...)

**Comment by stefan-niedermann** [comment]
@arcanemachine Which version exactly are you using ("About" section in the sidenav)?

@arcanemachine & @tecufanujacu can one of you create a test account for me so i can see exactly what's going on? The server seems to return an empty body on the requests.

**Comment by stefan-niedermann** [comment]
@arcanemachine & @tecufanujacu did you get the your versions from Play Store or from F-Droid?

**Comment by arcanemachine** [comment]
F-Droid

Was using version 2.2.0 when getting crashes

**Comment by stefan-niedermann** [comment]
Can you provide me a test account on your instance so i can analyse and debug the error?

**Comment by tecufanujacu** [comment]
@stefan-niedermann Nextcloud app and Notes app are both from Fdroid. I can't provide a test account on the machine in production, but if you need it I can replicate the istance on a dev machine, but for this i need a while.

**Comment by stefan-niedermann** [comment]
Sure, take your time. If you have any updates, i will be happy to try debug this issue.

**Comment by stefan-niedermann** [comment]
One more question: We heard that the (server) app "RansomwareDetection" causes issues wit SSO.

Is this app enabled abd does the problem persist if the app is disabled?

**Comment by arcanemachine** [comment]
I don't have the RansomwareDetection app.

Still getting crashes on 2.3.0 even after updating Nextcloud to latest stable version.

As mentioned before, my setup is LAN-only so I can't add a test account for you to use.

I think the problem is entirely local to the app as it crashes whenever I am unable to connect to the server, e.g. when I'm on a different WiFI network, or also when I restart the server when I _am_ on my home WiFi network.

Also, I only have WiFi sync enabled (since it's only accessible from my LAN).

Perhaps you could reproduce the issue by toggling on "WiFi sync only" and connecting to a WiFi network that has no functioning Internet access? Also, perhaps you could restart your server and see if there are any issues with your app while the server is restarting?

At any rate, the old version of the app is working great and I'm thankful for the awesome app.

**Comment by tecufanujacu** [comment]
> One more question: We heard that the (server) app "RansomwareDetection" causes issues wit SSO.
> 
> Is this app enabled abd does the problem persist if the app is disabled?

Me too, I don't use this app.

**Comment by tecufanujacu** [comment]
@stefan-niedermann, could the problem be related at apache/php versions?

Because I have the same identical installation of Nextcloud on two different machines but the problem is present only on one of them.

Production machine with:
Nextcloud 13.0.12
Apache 2.4.10
PHP 5.6

Dev machine:
Nextcloud 13.0.12
Apache 2.4.25
PHP 7

The problem is only present with the production machine, I noticed this just now while providing a test account for you in the Dev machine.

**Comment by stefan-niedermann** [comment]
@korelstar is PHP `5.6` still supported by Nextcloud (core) and the Notes app?

**Comment by tecufanujacu** [comment]
@stefan-niedermann, I also noticed that when I start the Notes app, it is able to sync the notes at least for one time, then if I retry to sync the notes it fails, but if I kill the app and I restart it, then it is newly able to sync at least one time.
I start the app and it is able to sync.
I retry to sync and the app fails with several errors.
I kill the app, I start the app and it is able to sync at least for another time.

**Comment by stefan-niedermann** [comment]
I just released v2.4.6 which comes with a new exception dialog. I hope that this will give us some better insights about what actually is happening.

If you want, you can directly install this zipped `.apk`-file for testing purposes.

[notes-2.4.6-(signed).apk.zip](https://github.com/stefan-niedermann/nextcloud-notes/files/4047941/notes-2.4.6-.signed.apk.zip)

**Comment by korelstar** [comment]
> Production machine with:
> Nextcloud 13.0.12
> PHP 5.6

All these components are out of support. Please update to supported versions! Latest stable versions are Nextcloud 17.0.2/16.0.7, PHP 7.4.1/7.3.13, Nextcloud Notes (Server App) 3.1.1. That's a huge difference!

**Comment by tecufanujacu** [comment]
> I just released v2.4.6 which comes with a new exception dialog. I hope that this will give us some better insights about what actually is happening.
> 
> If you want, you can directly install this zipped `.apk`-file for testing purposes.
> 
> [notes-2.4.6-(signed).apk.zip](https://github.com/stefan-niedermann/nextcloud-notes/files/4047941/notes-2.4.6-.signed.apk.zip)

Here the error:
```
App Version: 2.4.6
App Version Code: 69
Files App Version Code: 30090290

---

OS Version: 3.0.101-gff9641e(fd091a6552)
OS API Level: 23
Device: asanti_c
Model (and Product): PHOTON Q (XT897)

---

java.lang.ClassNotFoundException: org.apache.commons.httpclient.ProtocolException
	at java.lang.Class.classForName(Native Method)
	at java.lang.Class.forName(Class.java:324)
	at java.io.ObjectInputStream.resolveClass(ObjectInputStream.java:2258)
	at java.io.ObjectInputStream.readNewClassDesc(ObjectInputStream.java:1641)
	at java.io.ObjectInputStream.readClassDesc(ObjectInputStream.java:657)
	at java.io.ObjectInputStream.readNewObject(ObjectInputStream.java:1782)
	at java.io.ObjectInputStream.readNonPrimitiveContent(ObjectInputStream.java:761)
	at java.io.ObjectInputStream.readObject(ObjectInputStream.java:1983)
	at java.io.ObjectInputStream.readObject(ObjectInputStream.java:1940)
	at com.nextcloud.android.sso.api.AidlNetworkRequest.deserializeObjectV2(AidlNetworkRequest.java:318)
	at com.nextcloud.android.sso.api.AidlNetworkRequest.performNetworkRequestV2(AidlNetworkRequest.java:150)
	at com.nextcloud.android.sso.api.NextcloudAPI.performNetworkRequestV2(NextcloudAPI.java:148)
	at it.niedermann.owncloud.notes.persistence.NotesClient.requestServer(NotesClient.java:191)
	at it.niedermann.owncloud.notes.persistence.NotesClient.getNotes(NotesClient.java:123)
	at it.niedermann.owncloud.notes.persistence.NoteServerSyncHelper$SyncTask.pullRemoteChanges(NoteServerSyncHelper.java:362)
	at it.niedermann.owncloud.notes.persistence.NoteServerSyncHelper$SyncTask.doInBackground(NoteServerSyncHelper.java:290)
	at it.niedermann.owncloud.notes.persistence.NoteServerSyncHelper$SyncTask.doInBackground(NoteServerSyncHelper.java:261)
	at android.os.AsyncTask$2.call(AsyncTask.java:295)
	at java.util.concurrent.FutureTask.run(FutureTask.java:237)
	at android.os.AsyncTask$SerialExecutor$1.run(AsyncTask.java:234)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1113)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:588)
	at java.lang.Thread.run(Thread.java:818)
Caused by: java.lang.ClassNotFoundException: Didn't find class "org.apache.commons.httpclient.ProtocolException" on path: DexPathList[[zip file "/data/app/it.niedermann.owncloud.notes-1/base.apk"],nativeLibraryDirectories=[/data/app/it.niedermann.owncloud.notes-1/lib/arm, /vendor/lib, /system/lib]]
	at dalvik.system.BaseDexClassLoader.findClass(BaseDexClassLoader.java:56)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:511)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:469)
	... 23 more
	Suppressed: java.lang.ClassNotFoundException: org.apache.commons.httpclient.ProtocolException
		at java.lang.Class.classForName(Native Method)
		at java.lang.BootClassLoader.findClass(ClassLoader.java:781)
		at java.lang.BootClassLoader.loadClass(ClassLoader.java:841)
		at java.lang.ClassLoader.loadClass(ClassLoader.java:504)
		... 24 more
	Caused by: java.lang.NoClassDefFoundError: Class not found using the boot class loader; no stack trace available

```



> > Production machine with:
> > Nextcloud 13.0.12
> > PHP 5.6
> 
> All these components are out of support. Please update to supported versions! Latest stable versions are Nextcloud 17.0.2/16.0.7, PHP 7.4.1/7.3.13, Nextcloud Notes (Server App) 3.1.1. That's a huge difference!

I know, but in my case it isn't possible update at the moment and Notes is the only app that doesnt' work with SSO, for example I have no problem with DAVx⁵, but if the problem is caused by outdated software on server then I can stay for a while with an old Notes app.

**Comment by stefan-niedermann** [comment]
Photon Q, wow, that is an old device! I assume you are using LineageOS since you managed to run Android 6 on it? Apparently the communication between the Notes app and the Files app goes wrong because a class is missing. You should report this to the creator of your LineageOS-ROM, i suppose the bug there.

@tobiasKaminsky told me, that the Files app will ship `HttpClient` with version `3.10.x` manually, so this might change something on the situation (or it might not :man_shrugging: ). However, there is not much i can do, if this class is missing on your device.

Besides that: Nextcloud 13 has reached [end of life](https://github.com/nextcloud/server/wiki/Maintenance-and-Release-Schedule) **one year** ago! This is a security issue not only for you but for everyone else in the internet. You should *really* **really** your software.

**Comment by tecufanujacu** [comment]
> Photon Q, wow, that is an old device! I assume you are using LineageOS since you managed to run Android 6 on it? Apparently the communication between the Notes app and the Files app goes wrong because a class is missing. You should report this to the creator of your LineageOS-ROM, i suppose the bug there.
> 
> @tobiasKaminsky told me, that the Files app will ship `HttpClient` with version `3.10.x` manually, so this might change something on the situation (or it might not man_shrugging ). However, there is not much i can do, if this class is missing on your device.
> 
> Besides that: Nextcloud 13 has reached [end of life](https://github.com/nextcloud/server/wiki/Maintenance-and-Release-Schedule) **one year** ago! This is a security issue not only for you but for everyone else in the internet. You should _really_ **really** your software.

The Photon Q has Android 7 with LineageOS.

I have this error only when syncing with the problematic production machine.
I don't have this error when syncing with the production machine, in this case everythings works correctly, without any error.
I don't think that the error is in the roms otherwise it should happens always, with both the machines.

Seems to me that there is some sort of timeout or something else, in fact when the sync with the problematic machine has success (as reported in [573203174](https://github.com/stefan-niedermann/nextcloud-notes/issues/652#issuecomment-573203174)), the error in the Notes app doesn't appear and in apache access log I can read:

```
GET /nextcloud/index.php/apps/notes/api/v0.2/notes?pruneBefore=********* HTTP/1.1" 304 *** "-" "Mozilla/5.0 (Android) Nextcloud-android/3.9.2"
```

When instead the sync with the problematic machine fails, then in the Notes app I have the error and in apache log there isn't nothing.

Same problem with other two devices:
LG G2, Android 6 with LineageOS
OnePlus 5t, Android 9 with LineageOS

```
App Version: 2.4.6
App Version Code: 69
Files App Version Code: 30090290

---

OS Version: 3.4.111-LambdaKernel-Pure-Stable(8bd36fc591)
OS API Level: 23
Device: d802
Model (and Product): LG-D802 (cm_d802)

---

java.lang.ClassNotFoundException: org.apache.commons.httpclient.ProtocolException
	at java.lang.Class.classForName(Native Method)
	at java.lang.Class.forName(Class.java:324)
	at java.io.ObjectInputStream.resolveClass(ObjectInputStream.java:2258)
	at java.io.ObjectInputStream.readNewClassDesc(ObjectInputStream.java:1641)
	at java.io.ObjectInputStream.readClassDesc(ObjectInputStream.java:657)
	at java.io.ObjectInputStream.readNewObject(ObjectInputStream.java:1782)
	at java.io.ObjectInputStream.readNonPrimitiveContent(ObjectInputStream.java:761)
	at java.io.ObjectInputStream.readObject(ObjectInputStream.java:1983)
	at java.io.ObjectInputStream.readObject(ObjectInputStream.java:1940)
	at com.nextcloud.android.sso.api.AidlNetworkRequest.deserializeObjectV2(AidlNetworkRequest.java:318)
	at com.nextcloud.android.sso.api.AidlNetworkRequest.performNetworkRequestV2(AidlNetworkRequest.java:150)
	at com.nextcloud.android.sso.api.NextcloudAPI.performNetworkRequestV2(NextcloudAPI.java:148)
	at it.niedermann.owncloud.notes.persistence.NotesClient.requestServer(NotesClient.java:191)
	at it.niedermann.owncloud.notes.persistence.NotesClient.getNotes(NotesClient.java:123)
	at it.niedermann.owncloud.notes.persistence.NoteServerSyncHelper$SyncTask.pullRemoteChanges(NoteServerSyncHelper.java:362)
	at it.niedermann.owncloud.notes.persistence.NoteServerSyncHelper$SyncTask.doInBackground(NoteServerSyncHelper.java:290)
	at it.niedermann.owncloud.notes.persistence.NoteServerSyncHelper$SyncTask.doInBackground(NoteServerSyncHelper.java:261)
	at android.os.AsyncTask$2.call(AsyncTask.java:295)
	at java.util.concurrent.FutureTask.run(FutureTask.java:237)
	at android.os.AsyncTask$SerialExecutor$1.run(AsyncTask.java:234)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1113)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:588)
	at java.lang.Thread.run(Thread.java:818)
Caused by: java.lang.ClassNotFoundException: Didn't find class "org.apache.commons.httpclient.ProtocolException" on path: DexPathList[[zip file "/data/app/it.niedermann.owncloud.notes-1/base.apk"],nativeLibraryDirectories=[/data/app/it.niedermann.owncloud.notes-1/lib/arm, /vendor/lib, /system/lib]]
	at dalvik.system.BaseDexClassLoader.findClass(BaseDexClassLoader.java:56)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:511)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:469)
	... 23 more
	Suppressed: java.lang.ClassNotFoundException: Didn't find class "org.apache.commons.httpclient.ProtocolException" on path: DexPathList[[dex file "/data/dalvik-cache/xposed_XResourcesSuperClass.dex"],nativeLibraryDirectories=[/vendor/lib, /system/lib]]
		at dalvik.system.BaseDexClassLoader.findClass(BaseDexClassLoader.java:56)
		at java.lang.ClassLoader.loadClass(ClassLoader.java:511)
		at java.lang.ClassLoader.loadClass(ClassLoader.java:504)
		... 24 more
		Suppressed: java.lang.ClassNotFoundException: org.apache.commons.httpclient.ProtocolException
			at java.lang.Class.classForName(Native Method)
			at java.lang.BootClassLoader.findClass(ClassLoader.java:781)
			at java.lang.BootClassLoader.loadClass(ClassLoader.java:841)
			at java.lang.ClassLoader.loadClass(ClassLoader.java:504)
			... 25 more
		Caused by: java.lang.NoClassDefFoundError: Class not found using the boot class loader; no stack trace available

```




```
App Version: 2.4.6
App Version Code: 69
Files App Version Code: 30090290

---

OS Version: 4.4.153-lineage+(108a657d1b)
OS API Level: 28
Device: OnePlus5T
Model (and Product): ONEPLUS A5010 (OnePlus5T)

---

java.lang.ClassNotFoundException: org.apache.commons.httpclient.ProtocolException
	at java.lang.Class.classForName(Native Method)
	at java.lang.Class.forName(Class.java:453)
	at java.io.ObjectInputStream.resolveClass(ObjectInputStream.java:628)
	at java.io.ObjectInputStream.readNonProxyDesc(ObjectInputStream.java:1615)
	at java.io.ObjectInputStream.readClassDesc(ObjectInputStream.java:1520)
	at java.io.ObjectInputStream.readOrdinaryObject(ObjectInputStream.java:1776)
	at java.io.ObjectInputStream.readObject0(ObjectInputStream.java:1353)
	at java.io.ObjectInputStream.readObject(ObjectInputStream.java:373)
	at com.nextcloud.android.sso.api.AidlNetworkRequest.deserializeObjectV2(AidlNetworkRequest.java:318)
	at com.nextcloud.android.sso.api.AidlNetworkRequest.performNetworkRequestV2(AidlNetworkRequest.java:150)
	at com.nextcloud.android.sso.api.NextcloudAPI.performNetworkRequestV2(NextcloudAPI.java:148)
	at it.niedermann.owncloud.notes.persistence.NotesClient.requestServer(NotesClient.java:191)
	at it.niedermann.owncloud.notes.persistence.NotesClient.getNotes(NotesClient.java:123)
	at it.niedermann.owncloud.notes.persistence.NoteServerSyncHelper$SyncTask.pullRemoteChanges(NoteServerSyncHelper.java:362)
	at it.niedermann.owncloud.notes.persistence.NoteServerSyncHelper$SyncTask.doInBackground(NoteServerSyncHelper.java:290)
	at it.niedermann.owncloud.notes.persistence.NoteServerSyncHelper$SyncTask.doInBackground(NoteServerSyncHelper.java:261)
	at android.os.AsyncTask$2.call(AsyncTask.java:333)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at android.os.AsyncTask$SerialExecutor$1.run(AsyncTask.java:245)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
	at java.lang.Thread.run(Thread.java:764)
Caused by: java.lang.ClassNotFoundException: Didn't find class "org.apache.commons.httpclient.ProtocolException" on path: DexPathList[[zip file "/data/app/it.niedermann.owncloud.notes-FVRLtOccd74dZtm0KR8UYg==/base.apk"],nativeLibraryDirectories=[/data/app/it.niedermann.owncloud.notes-FVRLtOccd74dZtm0KR8UYg==/lib/arm64, /system/lib64]]
	at dalvik.system.BaseDexClassLoader.findClass(BaseDexClassLoader.java:134)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:379)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:312)
	... 22 more

```

About the fact that I should update, I know, but I can't to this before some few months.

**Comment by stefan-niedermann** [comment]
And all three phones work on the dev macbine but none of the on the productive?

Is there maybe some kind of TLS downgrade on your old server? E.g. does not support TLS 1.2, but says on the handshake "Hey client, i don't know what you want, but i can offer TLS 1.1“. Then the client might search for this implementation and doesn't find this old class...?

@tobiasKaminsky @AndyScherzinger @david-development

**Comment by stefan-niedermann** [comment]
Can you please install latest Dev-version of the files app from F-Droid and see what happens with that? (please uninstall files and notes previously, so you have a clean testing environment)

**Comment by tecufanujacu** [comment]
> And all three phones work on the dev macbine but none of the on the productive?
> 
> Is there maybe some kind of TLS downgrade on your old server? E.g. does not support TLS 1.2, but says on the handshake "Hey client, i don't know what you want, but i can offer TLS 1.1“. Then the client might search for this implementation and doesn't find this old class...?
> 
> @tobiasKaminsky @AndyScherzinger @David-Development

Yes, all three phones work on the dev machine but none of them on the productive.
About the TLS there isn't any problem on my old server, anyway to be sure I've just checked and it correctly uses the Protocol  : TLSv1.2.

I've also tried with the dev version of the Nexctloud client and the situation is the same.

What a strange problem.

**Comment by tecufanujacu** [comment]
@stefan-niedermann, I was been able to catch another log.
I open the app the first time and it is able to sync.
I go to the home leaving the app running in background.
I reopen the app and I have this new error:

```
App Version: 2.4.6
App Version Code: 69
Files App Version Code: com.nextcloud.client

---

OS Version: 3.0.101-gff9641e(fd091a6552)
OS API Level: 23
Device: asanti_c
Model (and Product): PHOTON Q (XT897)

---

java.lang.RuntimeException: Unable to resume activity {it.niedermann.owncloud.notes/it.niedermann.owncloud.notes.android.activity.NotesListViewActivity}: java.lang.IllegalArgumentException: Service not registered: com.nextcloud.android.sso.api.AidlNetworkRequest$1@2c36534
	at android.app.ActivityThread.performResumeActivity(ActivityThread.java:3137)
	at android.app.ActivityThread.handleResumeActivity(ActivityThread.java:3168)
	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2515)
	at android.app.ActivityThread.-wrap11(ActivityThread.java)
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1363)
	at android.os.Handler.dispatchMessage(Handler.java:102)
	at android.os.Looper.loop(Looper.java:148)
	at android.app.ActivityThread.main(ActivityThread.java:5461)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:726)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:616)
Caused by: java.lang.IllegalArgumentException: Service not registered: com.nextcloud.android.sso.api.AidlNetworkRequest$1@2c36534
	at android.app.LoadedApk.forgetServiceDispatcher(LoadedApk.java:1045)
	at android.app.ContextImpl.unbindService(ContextImpl.java:1347)
	at android.content.ContextWrapper.unbindService(ContextWrapper.java:622)
	at com.nextcloud.android.sso.api.AidlNetworkRequest.stop(AidlNetworkRequest.java:109)
	at com.nextcloud.android.sso.api.NextcloudAPI.stop(NextcloudAPI.java:80)
	at it.niedermann.owncloud.notes.persistence.NotesClient.updateAccount(NotesClient.java:99)
	at it.niedermann.owncloud.notes.persistence.NoteServerSyncHelper.updateAccount(NoteServerSyncHelper.java:141)
	at it.niedermann.owncloud.notes.android.activity.NotesListViewActivity.selectAccount(NotesListViewActivity.java:274)
	at it.niedermann.owncloud.notes.android.activity.NotesListViewActivity.onResume(NotesListViewActivity.java:228)
	at android.app.Instrumentation.callActivityOnResume(Instrumentation.java:1259)
	at android.app.Activity.performResume(Activity.java:6327)
	at android.app.ActivityThread.performResumeActivity(ActivityThread.java:3126)
	... 10 more

```

**Comment by stefan-niedermann** [comment]
@arcanemachine can you also post a stack trace from version 2.4.6 (or greater) here, so we can see whether the issue is the same like @tecufanujacu ?

**Comment by arcanemachine** [comment]
OK just waiting for the F-Droid repo to catch up.

**Comment by arcanemachine** [comment]
Not having any issues on 2.4.4 OR 2.4.6. Not sure what happened but everything is all good on my end (for now at least). Thanks a bunch.

**Comment by stefan-niedermann** [comment]
Thank you @arcanemachine . Please open a new issue if you have any problems.

**Comment by stefan-niedermann** [comment]
[Your last stacktrace](https://github.com/stefan-niedermann/nextcloud-notes/issues/652#issuecomment-573330199) was a known issue which has been fixed with version `2.4.8` and is not related to the other issue / stacktrace.

**Comment by stefan-niedermann** [comment]
@tecufanujacu to be honest, i have no idea what might be wrong here.

But considering that you are using way outdated software on your prod machine and that it works with supported versions besides the fact that i can't reproduce this, i will close this issue.

It is most likely some problem with your local setup.

Please feel free to report it again if you have deployed the supported versions from your dev environment and it still fails.

Sorry, but i simply cannot help you at this point. 😞
